app/controllers/application_controller.rb

```diff
@@ -187,7 +187,7 @@ def authorize_user_for_course
     end
 
     # check if course was disabled
-    if @course.disabled? && !@cud.has_auth_level?(:instructor)
+    if @course.is_disabled?
```

Comment:
You should add the `&& !@cud.has_auth_level?(:instructor)` because removing this will make the instructor unable to access the disable course.